SourceryRuntime/Sources/Definable.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+/// Describes that the object is defined in a context of some `Type`
+public protocol Defined: class {
+    /// Reference to type name where the object is defined, 
+    /// nil if defined outside of any `enum`, `struct`, `class` etc
+    var definedInTypeName: TypeName? { get }
+
+    /// Reference to actual type where the object is defined, 
+    /// nil if defined outside of any `enum`, `struct`, `class` etc or type is unknown
+    var definedInType: Type? { get }
+
+    /// Reference to actual type name where the method is defined if declaration uses typealias, otherwise just a `definedInTypeName`
+    var actualDefinedInTypeName: TypeName? { get }
```

Comment:
> non-'@objc' property 'actualDefinedInTypeName' does not satisfy requirement of '@objc' protocol 'MethodAutoJSExport'
    var actualDefinedInTypeName: TypeName? {
        ^
    @objc 
/Users/antondomashnev/Work/Sourcery/SourceryRuntime/Sources/JSExport.generated.swift:170:9: note: requirement 'actualDefinedInTypeName' declared here
    var actualDefinedInTypeName: TypeName? { get }

There is no extension for `Typed` it maybe the same eventually.